homeassistant/components/music_assistant/media_player.py

```diff
@@ -276,22 +276,36 @@ async def async_on_update(self) -> None:
             self._attr_state = MediaPlayerState(player.state.value)
         else:
             self._attr_state = MediaPlayerState(STATE_OFF)
+
         group_members_entity_ids: list[str] = []
+        coordinator_id: str | None = None
+        group_childs: set[str] = set()
         if player.group_childs:
+            # This player is the coordinator, with children players
+            coordinator_id = self.player_id
+            group_childs = set(player.group_childs) - {self.player_id}
+        elif player.synced_to:
+            # This player is a child player, with a coordinator
+            coordinator_id = player.synced_to
+            # Do not list siblings. It's unclear if we should list all siblings or just this player
+            group_childs = {self.player_id}
```

Comment:
this feels really weird - if you want to fill in the "group_members" attribute, then just give it the same list with the group leader always first.

Reply:
Yes, this was based on my own investigate of what other media players have done. I saw https://github.com/home-assistant/core/blob/220509fd6c55b9f0400539bdb7ffec536504ae04/homeassistant/components/bang_olufsen/media_player.py#L489-L506 only added one child without listing any siblings. But I can update this and it will simplify the code.